### PR TITLE
Cleanup party withdraw

### PR DESCRIPTION
**Pull request by cydh**
* Added enum for withdraw result.
* Added name of kicked player and changed `sd` init from `account_id` to `char_id`.
* Fixed party list window isn't updated when member kicked.

Signed-off-by: Cydh Ramdh <cydh@pservero.com>

---

TODO:
* I want add the leave/expel check through inter-server (but, too lazy)
* Support leaderless party

---
Reported by @aleos89

### CODE REVIEW

File: src/map/clif.h [hunk 1]
```diff
@@ -162,6 +162,13 @@ enum e_party_invite_reply {
 	PARTY_REPLY_INVALID_MAPPROPERTY_ME, ///< return=9 : !TODO "Cannot join a party in this map" -> MsgStringTable[1871] (since 20110205)
 };
 
+typedef enum PARTY_MEMBER_WITHDRAW {
```
Comment (Lemongrass3110) [inline src/map/clif.h]: Could you move it to mmo.h(so it can be used in the inter server too) and remove the typedef?

File: src/map/party.c [hunk 2]
```diff
@@ -604,29 +604,29 @@ int party_leave(struct map_session_data *sd)
...
 
 	party_trade_bound_cancel(sd);
-	intif_party_leave(p->party.party_id,sd->status.account_id,sd->status.char_id);
+	intif_party_leave(p->party.party_id,sd->status.account_id,sd->status.char_id,sd->status.name,PARTY_MEMBER_WITHDRAW_LEAVE);
 	return 1;
 }
 
 /// Invoked (from char-server) when a party member leaves the party.
-int party_member_withdraw(int party_id, uint32 account_id, uint32 char_id)
+int party_member_withdraw(int party_id, uint32 account_id, uint32 char_id, char *name, int type)
```
Comment (Lemongrass3110) [inline src/map/party.c]: force enum in the function definition would be better i guess?

File: src/char/int_party.c [hunk 3]
```diff
@@ -404,14 +404,16 @@ int mapif_party_optionchanged(int fd,struct party *p,uint32 account_id,int flag)
 }
 
 //Withdrawal notification party
-int mapif_party_withdraw(int party_id,uint32 account_id, uint32 char_id) {
-	unsigned char buf[16];
+int mapif_party_withdraw(int party_id, uint32 account_id, uint32 char_id, char *name, unsigned char type) {
+	unsigned char buf[16+NAME_LENGTH];
```
Comment (Lemongrass3110) [inline src/char/int_party.c]: Shouldnt that be 15?

**Comment by cydh** [comment]
sure

File: src/common/mmo.h [hunk 4]
```diff
@@ -838,6 +838,13 @@ enum e_pc_reg_loading {
 	PRL_ALL = 0xFF,
 };
 
+typedef enum PARTY_MEMBER_WITHDRAW {
```
Comment (secretdataz) [inline src/common/mmo.h]: Is typedef really neccessary for this?

File: src/common/mmo.h [hunk 5]
```diff
@@ -838,6 +838,13 @@ enum e_pc_reg_loading {
...
+	PARTY_MEMBER_WITHDRAW_LEAVE,	  ///< /leave
+	PARTY_MEMBER_WITHDRAW_EXPEL,	  ///< Kicked
+	PARTY_MEMBER_WITHDRAW_CANT_LEAVE, ///< TODO: Cannot /leave
+	PARTY_MEMBER_WITHDRAW_CANT_EXPEL, ///< TODO: Cannot be kicked
```
Comment (Lemongrass3110) [inline src/common/mmo.h]: Could that be the correct return for clif_parse_RemovePartyMember when you are on a party locked map?
While I looked this up: Should we not check the target player's map too?

File: src/common/mmo.h [hunk 6]
```diff
@@ -838,6 +838,13 @@ enum e_pc_reg_loading {
 	PRL_ALL = 0xFF,
 };
 
+typedef enum PARTY_MEMBER_WITHDRAW {
+	PARTY_MEMBER_WITHDRAW_LEAVE,	  ///< /leave
+	PARTY_MEMBER_WITHDRAW_EXPEL,	  ///< Kicked
+	PARTY_MEMBER_WITHDRAW_CANT_LEAVE, ///< TODO: Cannot /leave
```
Comment (Lemongrass3110) [inline src/common/mmo.h]: Could that be the correct return for clif_parse_LeaveParty when you are on a party locked map?
But that might be PACKETVER dependent.

Comment (cydh) [inline src/common/mmo.h]: I believe those enum are exists from ancient Aegis, maybe.
By the name, yes

File: src/common/mmo.h (hunk 5 above)
Comment (cydh) [inline src/common/mmo.h]: By the name, yes.

File: src/char/int_party.c [hunk 7]
```diff
@@ -404,14 +404,16 @@ int mapif_party_optionchanged(int fd,struct party *p,uint32 account_id,int flag)
...
+int mapif_party_withdraw(int party_id, uint32 account_id, uint32 char_id, char *name, PARTY_MEMBER_WITHDRAW type) {
+	unsigned char buf[15+NAME_LENGTH];
 
 	WBUFW(buf,0) = 0x3824;
 	WBUFL(buf,2) = party_id;
 	WBUFL(buf,6) = account_id;
 	WBUFL(buf,10) = char_id;
-	chmapif_sendall(buf, 14);
+	memcpy(WBUFP(buf,14), name, NAME_LENGTH);
+	WBUFB(buf,14+NAME_LENGTH) = type;
```
Comment (cydh) [inline src/char/int_party.c]: not significant

**Comment by MikeeMaturan** [comment]
[SQL]: DB error - Table 'ragnarok.log' doesn't exist
[Debug]: at ..\src\login\loginlog.c:88 - INSERT INTO `log`(`time`,`ip`,`user`,`r
code`,`log`) VALUES (NOW(), '0.0.0.0', 'login server', '100', 'login server star
ted')
[Info]: Connection request of the char-server 'rAthena' @ 127.0.0.1:6121 (accoun
t: 'mikee', pass: 'maturan', ip: '127.0.0.1')
[SQL]: DB error - Table 'ragnarok.log' doesn't exist
[Debug]: at ..\src\login\loginlog.c:88 - INSERT INTO `log`(`time`,`ip`,`user`,`r
code`,`log`) VALUES (NOW(), '127.0.0.1', 'mikee', '100', 'charserver - rAthena@1
27.0.0.1:6121')

[SQL]: DB error - Table 'ragnarok.item_db_re' doesn't exist
[Debug]: at ..\src\map\script.c:16132 - SELECT DISTINCT LEFT( `name_japanese`, 1
 ) AS alphabets FROM `item_db_re` RIGHT JOIN `mob_db_re` ON `item_db_re`.`id` =
`mob_db_re`.`dropcardid` WHERE ~(`MODE`) & 32 AND `type` = 6 GROUP BY `name_japa
nese` ORDER BY alphabets;

how to repair that issue?????

**Pull request merged**